File: Cpp/2103_Rings_and_Rods.cpp
class Solution {
public:
    int countPoints(string rings) {
        int result = 0;
        vector<unordered_set<char>> count(10);

        for(int i = 0; i < rings.size(); i += 2) {
            count[rings[i + 1] - '0'].insert(rings[i]);
        }

        for(auto& s : count) {
            if(s.size() == 3) ++result;
        }

        return result;
    }
};

File: Kotlin/2103_Rings_and_Rods.kt
class Solution {
    fun countPoints(rings: String): Int {
        val m = HashMap<Char, HashSet<Char>>()

        for (i in 0 until rings.length step 2) {
            val s = m.getOrDefault(rings[i + 1], hashSetOf())
            s.add(rings[i])
            m.put(rings[i + 1], s)
        }

        return m.values.count { it.size == 3}
    }
}